### PR TITLE
fix: Checks for error after rows.Next call

### PR DESCRIPTION
**Pull request by cliedeman**
Fix for #477

### CODE REVIEW

File: entc/gen/template/dialect/sql/globals.tmpl [hunk 1]
```diff
@@ -18,6 +18,9 @@ func insertLastID(ctx context.Context, tx dialect.Tx, insert *sql.InsertBuilder)
 		}
 		defer rows.Close()
 		if !rows.Next() {
+			if rows.Err() != nil {
+				return 0, rows.Err()
+			}
 			return 0, fmt.Errorf("no rows found for query: %v", query)
```
Comment (a8m) [inline entc/gen/template/dialect/sql/globals.tmpl:24]: You can remove this change and re-run codegen, I just deleted this template code.

File: dialect/sql/scan.go [hunk 2]
```diff
@@ -29,6 +30,10 @@ func ScanOne(rows ColumnScanner, v interface{}) error {
 		return fmt.Errorf("sql/scan: unexpected number of columns: %d", n)
 	}
 	if !rows.Next() {
+		if rows.Err() != nil {
+			return rows.Err()
+		}
+
```
Comment (a8m) [inline dialect/sql/scan.go:36]: remove this \n

**Comment by cliedeman** [comment]
@a8m updated

**Comment by a8m** [comment]
Thanks @cliedeman

**Pull request merged**